tools/system/dmf-get/main.c

```diff
@@ -781,6 +781,34 @@ static int DownloadModule(const char* module_name, const char* module_version,
         final_module_path[sizeof(final_module_path) - 1] = '\0';
     }
     
+    // Verify architecture matches expected architecture (if specified)
+    if (!already_exists && arch_name != NULL && arch_name[0] != '\0') {
+        char package_arch[DMOD_MAX_ARCH_NAME_LENGTH];
+        if (Dmod_GetPackageArchitecture(final_module_path, package_arch, sizeof(package_arch))) {
+            // Compare with the expected architecture
+            if (strcmp(package_arch, arch_name) != 0) {
+                DMOD_LOG_ERROR("Architecture mismatch: package is '%s', expected '%s'\n", 
+                       package_arch, arch_name);
+                DMOD_LOG_ERROR("Removing incompatible module file: %s\n", final_module_path);
+                
+                // Delete the incompatible file
+                remove(final_module_path);
+                
+                // Also remove .dmd file if it exists
+                if (dmd_file_path[0] != '\0') {
+                    remove(dmd_file_path);
+                }
+                
+                DMOD_LOG_ERROR("Error: Architecture mismatch - module not installed\n");
+                return 1;
+            } else {
+                DMOD_LOG_INFO("Architecture verified: %s\n", package_arch);
+            }
+        } else {
+            DMOD_LOG_INFO("Warning: Could not verify architecture (may be an older module format)\n");
+        }
+    }
+    
     // Process dependencies if requested and not already processed
     if (download_dependencies && !already_exists) {
```

Comment:
jak użytkownik nie poda architektury, to domyślna architektura to architektura systemu - czyli wtedy mamy `arch_name=DMOD_ARCH` i wtedy też sprawdzamy czy się zgadza w każdej ściągniętej paczce